--- local/target/linux/qualcommbe/ipq95xx/base-files/etc/board.d/02_network
+++ glinet/target/linux/qualcommbe/ipq95xx/base-files/etc/board.d/02_network
@@ -14,9 +14,6 @@
 	8devices,kiwi-dvk)
 		ucidef_set_interfaces_lan_wan "lan1 lan2" "wan"
 		;;
-	askey,sbe1v1k)
-		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
-		;;
 	qcom,ipq9574-ap-al02-c7)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4 lan5" "wan"
 		;;
@@ -26,25 +23,9 @@
 	esac
 }
 
-DEV_PATH_PCIE0=soc@0/10000000.pcie/pci0001:00/0001:00:00.0/0001:01:00.0
-DEV_PATH_PCIE1=soc@0/18000000.pcie/pci0003:00/0003:00:00.0/0003:01:00.0
-DEV_PATH_PCIE2=soc@0/20000000.pcie/pci0002:00/0002:00:00.0/0002:01:00.0
-
-ipq95xx_setup_wlan()
-{
-	local board="$1"
-
-	case "$board" in
-	askey,sbe1v1k)
-		ucidef_add_wlan "$DEV_PATH_PCIE0" "$DEV_PATH_PCIE1" "$DEV_PATH_PCIE2"
-		;;
-	esac
-}
-
 board_config_update
 board=$(board_name)
 ipq95xx_setup_interfaces $board
-ipq95xx_setup_wlan $board
 board_config_flush
 
 exit 0
